packages/instrumented-component-lifecycle/lib/instrumented-component-lifecycle.tsx

```diff
@@ -0,0 +1,78 @@
+import BugsnagPerformance from '@bugsnag/browser-performance'
+import { DISCARD_END_TIME } from '@bugsnag/core-performance'
+import type { Span } from '@bugsnag/core-performance'
+import * as React from 'react'
+import type { PropsWithChildren } from 'react'
+
+interface InstrumentedComponentLifecycleProps extends PropsWithChildren {
+  name: string
+  includeComponentUpdates: boolean
+}
+
+class InstrumentedComponentLifecycle extends React.Component<InstrumentedComponentLifecycleProps> {
+  private componentMountSpan: Span | undefined
+  private componentUpdateSpan: Span | undefined
+  private componentUnmountSpan: Span | undefined
+  private componentLifetimeSpan: Span | undefined
+
+  public static defaultProps: Partial<InstrumentedComponentLifecycleProps> = {
+    includeComponentUpdates: true
+  }
+
+  public constructor (props: InstrumentedComponentLifecycleProps) {
+    super(props)
+    const { name } = this.props
+
+    this.componentLifetimeSpan = BugsnagPerformance.appState !== 'ready' ? BugsnagPerformance.startSpan(`[ViewLoad/Component]${name}`) : undefined
+    this.componentMountSpan = BugsnagPerformance.appState !== 'ready' ? BugsnagPerformance.startSpan(`[ViewLoadPhase/Mount]${name}`) : undefined
+  }
+
+  async componentDidMount () {
+    if (this.componentMountSpan && BugsnagPerformance.appState === 'ready') {
+      this.componentMountSpan.end()
+    }
+  }
+
+  public shouldComponentUpdate ({ includeComponentUpdates = true }: InstrumentedComponentLifecycleProps, nextProps: any): boolean {
+    if (includeComponentUpdates && this.componentMountSpan && nextProps !== this.props.children) {
+      this.componentUpdateSpan = BugsnagPerformance.startSpan(`[ViewLoadPhase/Update]${this.props.name}`)
+    }
```

Comment:
if we're able to detect the props that changed here, we should also set a custom attribute. `bugsnag.component.update.props`